modules/dxp/apps/portal-workflow/portal-workflow-kaleo-designer-web/src/test/java/com/liferay/portal/workflow/kaleo/designer/web/internal/portlet/display/context/KaleoDesignerDisplayContextTest.java

```diff
@@ -0,0 +1,193 @@
+import com.liferay.portal.kernel.security.permission.resource.PortletResourcePermission;
+import com.liferay.portal.kernel.service.UserLocalService;
+import com.liferay.portal.kernel.test.util.RandomTestUtil;
+import com.liferay.portal.kernel.util.Portal;
+import com.liferay.portal.kernel.util.PortalUtil;
+import com.liferay.portal.security.script.management.configuration.helper.ScriptManagementConfigurationHelper;
+import com.liferay.portal.test.rule.LiferayUnitTestRule;
+import com.liferay.portal.workflow.kaleo.model.KaleoDefinitionVersion;
+import com.liferay.portal.workflow.kaleo.runtime.action.ActionExecutorManager;
+import com.liferay.portal.workflow.kaleo.service.KaleoDefinitionVersionLocalService;
+
+import java.text.SimpleDateFormat;
+
+import java.util.Collections;
+import java.util.Date;
+
+import javax.portlet.RenderRequest;
+
+import javax.servlet.http.HttpServletRequest;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+
+import org.mockito.Mockito;
+
+/**
+ * @author Nathaly Gomes
+ */
+public class KaleoDesignerDisplayContextTest {
+
+	@ClassRule
+	@Rule
+	public static final LiferayUnitTestRule liferayUnitTestRule =
+		LiferayUnitTestRule.INSTANCE;
+
+	@Before
+	public void setUp() {
+		_setUpKaleoDesignerDisplayContext();
+		_setUpPortalUtil();
+	}
+
+	@Test
+	public void testGetKaleoDefinitionVersionsJSONArray() throws Exception {
+		JSONArray expectedJSONArray = _expectedJSONArray(StringPool.BLANK);
+
+		JSONArray actualJSONArray =
+			_kaleoDesignerDisplayContext.getKaleoDefinitionVersionsJSONArray(
+				_kaleoDefinitionVersion);
+
+		Assert.assertEquals(
+			expectedJSONArray.toString(), actualJSONArray.toString());
+
+		User user = Mockito.mock(User.class);
+
+		Mockito.when(
+			_userLocalService.fetchUser(_kaleoDefinitionVersion.getUserId())
+		).thenReturn(
+			user
+		);
+
+		Mockito.when(
+			user.getFullName()
+		).thenReturn(
+			"Test Test"
+		);
+
+		expectedJSONArray = _expectedJSONArray("Test Test");
+
+		actualJSONArray =
+			_kaleoDesignerDisplayContext.getKaleoDefinitionVersionsJSONArray(
+				_kaleoDefinitionVersion);
+
+		Assert.assertEquals(
+			expectedJSONArray.toString(), actualJSONArray.toString());
+
+		Mockito.when(
+			_userLocalService.fetchUser(_kaleoDefinitionVersion.getUserId())
+		).thenReturn(
+			null
+		);
+
+		Mockito.when(
+			_userLocalService.fetchUserByScreenName(
+				_kaleoDefinitionVersion.getCompanyId(),
+				"default-service-account")
+		).thenReturn(
+			user
+		);
+
+		Mockito.when(
+			user.getFullName()
+		).thenReturn(
+			"default-service-account default-service-account"
+		);
+
+		expectedJSONArray = _expectedJSONArray(
+			"default-service-account default-service-account");
+
+		actualJSONArray =
+			_kaleoDesignerDisplayContext.getKaleoDefinitionVersionsJSONArray(
+				_kaleoDefinitionVersion);
+
+		Assert.assertEquals(
+			expectedJSONArray.toString(), actualJSONArray.toString());
+	}
```

Comment:
@vngoliveira try to divide this test into 3 parts by using comments, like this:

1. // Existing creator
2. // Removed creator
3. // Removed creator with no default-service-account